adserver/tasks.py

```diff
@@ -218,6 +233,105 @@ def daily_update_keywords(day=None):
                 )
 
 
+@app.task()
+def daily_update_regiontopic(day=None):  # pylint: disable=too-many-branches
+    """
+    Update the RegionTopicImpression index each day.
+
+    Each data point will have one region, but multiple possible topics.
+
+    :arg day: An optional datetime object representing a day
+    """
+    start_date, end_date = _get_day(day)
+
+    log.info("Updating RegionTopic's for %s-%s", start_date, end_date)
+
+    # Remove all old impressions, because they are cumulative
+    RegionTopicImpression.objects.filter(
+        date__gte=start_date,
+        date__lt=end_date,
+    ).delete()
+
+    for impression_type in IMPRESSION_TYPES:
+        queryset = _default_filters(impression_type, start_date, end_date)
+
+        topic_mapping = defaultdict(int)
+
+        for values in (
+            queryset.values("keywords", "country")
+            .annotate(total=Count("country"))
+            .filter(total__gt=0)
+            .order_by("-total")
+            .values(
+                "keywords",
+                "country",
+                "total",
+            )
+        ):
+            if not (values["keywords"] and values["country"]):
+                continue
+
+            keywords = json.loads(values["keywords"])
+            country = values["country"]
+            publisher_keywords = set(keywords)
+
+            topics = set()
+            for keyword in publisher_keywords:
+                if keyword in data_science:
+                    topic = "data-science"
+                elif keyword in backend_web:
+                    topic = "backend-web"
+                elif keyword in frontend_web:
+                    topic = "frontend-web"
+                elif keyword in security_privacy:
+                    topic = "security-privacy"
+                elif keyword in devops:
+                    topic = "devops"
+                elif keyword in python:
+                    topic = "python"
+                elif keyword in blockchain:
+                    topic = "blockchain"
+                elif keyword in game_dev:
+                    topic = "game-dev"
+                else:
+                    continue
```

Comment:
We are _not_ counting views that have no keywords here. I wasn't sure if we wanted an `other` topic. I'm guessing it will be most views (because it would include every junk keyword as `other`)

Reply:
I think that's OK actually. However, an `other` might be decent too.